Sources/JavaScriptKit/JS Types/JSPromise.swift

```diff
@@ -0,0 +1,295 @@
+//
+//  JSPromise.swift
+//  
+//
+//  Created by Alsey Coleman Miller on 6/3/20.
+//
+
+/**
+ JavaScript Promise
+ 
+ A Promise is a proxy for a value not necessarily known when the promise is created. It allows you to associate handlers with an asynchronous action's eventual success value or failure reason. This lets asynchronous methods return values like synchronous methods: instead of immediately returning the final value, the asynchronous method returns a promise to supply the value at some point in the future.
+ */
+public final class JSPromise<Success>: JSType where Success: JSValueConvertible, Success: JSValueConstructible {
```

Comment:
I think generic parameter `Success` doesn't match JavaScript's Promise use.

JavaScript's Promise allows users to return any type of values, so the requirement of single static type in JSPromise is too hard constraint.